apisix/plugins/cas-auth.lua

```diff
@@ -87,6 +89,9 @@ local function with_session_id(conf, ctx, session_id)
     else
         -- refresh the TTL
         store:set(session_id, user, SESSION_LIFETIME)
+        if conf.set_user_header then
```

Comment:
the `set_user_header` field is not needed, this feature can be implemented using the `user_header` field alone. Also please choose a better name to allow any header to be passed.

eg: field name: `send_headers` and the type of this field should be array, to allow multiple headers to be sent.

Reply:
I'm fine with removing `set_user_header`, and having `user_header` empty by default.

As far as `send_headers` is concerned, I'm not sure I understand what you mean.
Do you mean allowing the plugin to put in a header any value from the cas response ? Because that would require a lot more work as currently only the user is stored.
Or do you mean having a general array which we can search in in the code when other headers are implemented ? because that would remove the ability to overwrite the headers names, which would make it painful to integrate with some apps.

Reply:
What about the `hash/mapping` way as following:

```
{
    "XPATH_such_as: /auth/user/node": "X-Forwarded-User", // could be the default setting
    "other_xpath_node_to_extract_value": "X-Forwarded-Foo" 
}
```

Some related Lua libs:
- https://github.com/basiliscos/lua-xpath
- https://github.com/bkersten/lua-xpath

Reply:
> Because that would require a lot more work as currently only the user is stored.

Why would it require a lot more work? I don't understand you. If any of the headers configured in `send_headers` is not present in the request then the plugin won't add it. Isn't this simple?

Reply:
@shreemaan-abhishek I still don't understand what you mean.

For this part:

> the set_user_header field is not needed, this feature can be implemented using the user_header field alone. Also please choose a better name to allow any header to be passed.

I agree.

But for this part:

> Why would it require a lot more work? I don't understand you. If any of the headers configured in send_headers is not present in the request then the plugin won't add it. Isn't this simple?

I have no understanding of what you mean. If send_headers is not present in which request ? The CAS response ? 

I feel like we're not talking about the same thing. The goal of this change is to allow an upstream service to retrieve a value already stored: the username of the authenticated user.

Yes it could be made more generic, by storing more values from the cas response body:

> ngx_re_match(res.body, "<cas:user>(.*?)</cas:user>", "jo");

But it's currently an issue for many users, and it can be improved afterwards to allow for any value for the response.

Other authentication plugins on Apisix have a user_header parameter to pass the authenticated user to upstream services, why does this have to be different ?

Reply:
> Other authentication plugins on Apisix have a user_header parameter to pass the authenticated user to upstream services, why does this have to be different ?

can you show an example?

Reply:
never mind, I understood what you are trying to do.

> I'm fine with removing set_user_header, and having user_header empty by default.

please push this change then we can resolve this conversation.

Reply:
@shreemaan-abhishek done